sdk/batch/Azure.ResourceManager.Batch/tests/Helpers/ResourceDataHelper.cs

```diff
@@ -0,0 +1,178 @@
+            Assert.AreEqual(account1.Location, account2.Location);
+            Assert.AreEqual(account1.AccountEndpoint, account2.AccountEndpoint);
+            Assert.AreEqual(account1.Name, account2.Name);
+        }
+        #endregion
+
+        #region Pool
+        public static BatchAccountPoolData GetBatchAccountPoolData()
+        {
+            var data = new BatchAccountPoolData()
+            {
+                DisplayName = "test_pool",
+                VmSize = "small",
+                DeploymentConfiguration = new BatchDeploymentConfiguration()
+                {
+                    CloudServiceConfiguration = new BatchCloudServiceConfiguration("2")
+                },
+                StartTask = new BatchAccountPoolStartTask()
+                {
+                    CommandLine = "cmd.exe /c \"echo hello world\"",
+                    ResourceFiles =
+                    {
+                        new BatchResourceFile()
+                        {
+                            HttpUri = new Uri("https://blobsource.com"),
+                            FilePath = "filename.txt",
+                            Identity = new ComputeNodeIdentityReference()
+                            {
+                                ResourceId = new ResourceIdentifier("refUserId123")
+                            }
+                        }
+                    },
+                    EnvironmentSettings =
+                    {
+                        new BatchEnvironmentSetting("ENV_VAR")
+                        {
+                            Value = "env_value"
+                        }
+                    },
+                    UserIdentity = new BatchUserIdentity()
+                    {
+                        AutoUser = new BatchAutoUserSpecification()
+                        {
+                            ElevationLevel = BatchUserAccountElevationLevel.Admin,
+                        }
+                    }
+                },
+                /*UserAccounts =
+                {
+                    //new BatchUserAccount("adminUser", "xyz123"),
+                    new BatchUserAccount("testaccount", "randompasswd")
+                },*/
+                ScaleSettings = new BatchAccountPoolScaleSettings()
+                {
+                    FixedScale = new BatchAccountFixedScaleSettings()
+                    {
+                        TargetDedicatedNodes = 0,
+                        TargetLowPriorityNodes = 0
+                    }
+                }
+            };
+            return data;
+        }
+        public static void AssertPoolData(BatchAccountPoolData poolData1, BatchAccountPoolData poolData2)
+        {
+            AssertResourceData(poolData1, poolData2);
+            Assert.AreEqual(poolData1.ETag, poolData2.ETag);
+            Assert.AreEqual(poolData1.AllocationState, poolData2.AllocationState);
+            Assert.AreEqual(poolData1.SystemData, poolData2.SystemData);
+        }
```

Comment:
Only an example, do not assert SystemData
```suggestion
            AssertSystemData(poolData1.SystemData, poolData2.SystemData);
        }
        
        public static void AssertSystemData(SystemData data1, SystemData data2)
        {
        Assert.AreEqual(data1.prop1, data2.prop1);
        // ... etc
        }
```

Reply:
fixed